compiler/ir/ir.psi2ir/src/org/jetbrains/kotlin/psi2ir/generators/StatementGenerator.kt

```diff
@@ -131,40 +135,54 @@ class StatementGenerator(
 
     override fun visitDestructuringDeclaration(multiDeclaration: KtDestructuringDeclaration, data: Nothing?): IrStatement {
         val irBlock = IrCompositeImpl(
-            multiDeclaration.startOffsetSkippingComments, multiDeclaration.endOffset,
+            UNDEFINED_OFFSET, UNDEFINED_OFFSET,
             context.irBuiltIns.unitType, IrStatementOrigin.DESTRUCTURING_DECLARATION
         )
         val ktInitializer = multiDeclaration.initializer!!
-        val containerValue = scope.createTemporaryVariableInBlock(context, generateExpression(ktInitializer), irBlock, "container")
+        val irInitializer = generateExpression(ktInitializer)
 
-        declareComponentVariablesInBlock(multiDeclaration, irBlock, containerValue)
+        val containerVariable = scope.declareTemporaryVariableInBlock(irInitializer, irBlock, nameHint = "container")
+
+        declareComponentVariablesInBlock(multiDeclaration, irBlock) { containerValueWithOffsets ->
+            if (containerValueWithOffsets) {
+                VariableLValue(context, containerVariable)
+            } else {
+                VariableLValue(context, containerVariable, startOffset = UNDEFINED_OFFSET, endOffset = UNDEFINED_OFFSET)
+            }
+        }
 
         return irBlock
     }
 
     fun declareComponentVariablesInBlock(
         multiDeclaration: KtDestructuringDeclaration,
         irBlock: IrStatementContainer,
-        containerValue: IntermediateValue
+        containerValueWithOffsets: (Boolean) -> IntermediateValue
```

Comment:
I'd prefer to pass two `IntermediateValue`s here, something like `firstContainerValue: IntermediateValue, restContainerValue: IntermediateValue`.

Reply:
Done!